ravenpy/models/commands.py

```diff
@@ -179,17 +204,395 @@ class GriddedForcingCommand(NamedTuple):
+        :Units,mm,mm,mm,mm,mm,mm,mm,mm,C,mm,0-1,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm,mm
+        {hru_states}
+    :EndHRUStateVariableTable
+    """
+    hru_states: Dict[int, HRUStateVariableTableCommandRecord] = field(
+        default_factory=dict
+    )
+
+    def to_rv(self):
+        return dedent(self.template).format(
+            hru_states="\n".join(map(str, self.hru_states.values()))
+        )
+
+
+@dataclass
+class BasinIndexCommand(RavenConfig):
+    """Initial conditions for a flow segment."""
+
+    template = """
+    :BasinIndex {index},{name}
+        :ChannelStorage, {channelstorage}
+        :RivuletStorage, {rivuletstorage}
+        :Qout,{nsegs},{qout},{qoutlast}
+        :Qlat,{nQlatHist},{qlat},{qlatlast}
+        :Qin ,{nQinHist}, {qin}
+        """
+
+    index: int = 1
+    name: str = "watershed"
+    channelstorage: float = 0
+    rivuletstorage: float = 0
+    qout: tuple = (0,)
+    qoutlast: float = 0
+    qlat: tuple = (0, 0, 0)
+    qlatlast: float = 0
+    qin: tuple = 20 * (0,)
+
+    def to_rv(self):
+        return (
+            dedent(self.template)
+            .format(
+                **asdict(self),
+                nsegs=len(self.qout),
+                nQlatHist=len(self.qlat),
+                nQinHist=len(self.qin),
+            )
+            .replace("(", "")
+            .replace(")", "")
+        )
+
+
+@dataclass
+class BasinStateVariablesCommand(RavenConfig):
+
+    template = """
+    :BasinStateVariables
+        {basin_states_list}
+    :EndBasinStateVariables
+    """
+
+    basin_states: Dict[int, BasinIndexCommand] = field(default_factory=dict)
+
+    def to_rv(self):
+        return dedent(self.template).format(
+            basin_states_list="\n".join(map(str, self.basin_states.values()))
+        )
+
+
+@dataclass
+class SoilClassesCommand(RavenConfig):
+    @dataclass
+    class Record(RavenConfig):
```

Comment:
@huard This is my new idea of having an internal `dataclass` to encapsulate the records that I was telling you about, instead of having a top-level class. I think it makes sense and is much cleaner, but I will wait for your input before updating the previous ones.